test/core/store/fs/impl/test_cog.py

```diff
@@ -0,0 +1,193 @@
+# The MIT License (MIT)
+# Copyright (c) 2021/2022 by the xcube team and contributors
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+import os.path
+import unittest
+
+import fsspec
+import rasterio as rio
+import rioxarray
+import s3fs
+import xarray
+import xarray as xr
+
+# from xcube.core.store.fs.impl import cog
+# from test.s3test import S3Test, MOTO_SERVER_ENDPOINT_URL
+from xcube.core.store.fs.impl.cog import GeoTIFFMultiLevelDataset, \
+    MultiLevelDatasetGeoTiffFsDataAccessor
+from xcube.core.store.fs.impl.dataset import DatasetGeoTiffFsDataAccessor
+from xcube.util.jsonschema import JsonObjectSchema
+
+
+class RioXarrayTest(unittest.TestCase):
```

Comment:
Please add a docstring that justifies why we have this test. Say, that it doesn't test xcube but asserts that `rioxarray` works as we expect it.